[Alerting] Preconfigured alert history index connector (#94909)
elastic/kibana: docs/user/alerting/action-types/pre-configured-connectors.asciidoc

[role="screenshot"]
image::images/pre-configured-connectors-view-screen.png[Pre-configured connector view details]

[float]
[[preconfigured-connector-alert-history]]

> ==== Alert History Preconfigured Connector

Reviewer: ```suggestion
==== Alert history preconfigured connector
```

{kib} offers a preconfigured <<index-action-type, Index>> connector to facilitate indexing active alert data into {es}.

To use this connector, set the <<action-settings, `xpack.actions.preconfiguredAlertHistoryEsIndex`>> configuration to `true`. 

```js

>   xpack.actions.preconfiguredAlertHistoryEsIndex: true

Author: Updated prefix to `kibana-alert-history-` in https://github.com/elastic/kibana/pull/94909/commits/095b495f5244b28ab513df8bc89a9c3903039c52

```

> When creating a new rule, add an <<index-action-type, Index action>> and select the `Alert History ES Index (preconfigured)` connector.

Reviewer: This link confuses me.  It takes me to the Index connector and action page.  I'm still not sure what the actions are and how it relates to the screenshot.

Author: This is a type of index connector, which is why I added the link to the index connector type. Normally with an index connector, the user who configures the connector decides what the index name is and the user who adds the connector to a rule decides what the data schema is. For this preconfigured alert history connector, the index name and data schema is pre-determined.

[role="screenshot"]
image::images/pre-configured-alert-history-connector.png[Select pre-configured alert history connectors]

> With this connector, documents are indexed using a pre-configured schema that captures all of the rule and alert information that is available as <<defining-alerts-actions-variables, action variables>>. By default, these documents will be indexed into the `alert-history-default` index, but you may specify a different index if desired. Note that all index names must start with `alert-history-` in order to take advantage of the pre-configured Alert History index template.

Reviewer: ```suggestion
Documents are indexed using a preconfigured schema that captures the available rule and alert information as <<defining-alerts-actions-variables, action variables>>. By default, these documents are indexed into the `alert-history-default` index, but you can specify a different index. To take advantage of the preconfigured Alert History index template, start index names with `alert-history-`.
```

[IMPORTANT]
==============================================

> Make sure you have `all` or `write` privileges to the `alert-history-*` indices in order to ensure your documents will be written. Refer to <<xpack-kibana-role-management>> for more information.

Reviewer: ```suggestion
To write to documents, you must have `all` or `write` privileges to the `alert-history-*` indices. Refer to <<xpack-kibana-role-management>> for more information.
```

Reviewer: Not sure about this edit, but I think you can tighten this sentence.
